src/ripple/rpc/impl/Handler.cpp

```diff
@@ -205,36 +245,47 @@ class HandlerTable
             version > (betaEnabled ? RPC::apiBetaVersion
                                    : RPC::apiMaximumSupportedVersion))
             return nullptr;
-        auto i = table_.find(name);
-        return i == table_.end() ? nullptr : &i->second;
+
+        auto const range = table_.equal_range(name);
+        auto const i = std::find_if(
+            range.first, range.second, [version](auto const& entry) {
+                return version >= entry.second.minApiVer_ &&
+                    version <= entry.second.maxApiVer_;
+            });
+
+        return i == range.second ? nullptr : &i->second;
     }
 
     std::vector<char const*>
     getHandlerNames() const
     {
         std::vector<char const*> ret;
-        ret.reserve(table_.size());
         for (auto const& i : table_)
-            ret.push_back(i.second.name_);
+        {
+            // Note, table_ is always ordered, allowing such a simple check
+            if (ret.empty() || std::strcmp(ret.back(), i.second.name_) != 0)
+                ret.push_back(i.second.name_);
+        }
+
         return ret;
     }
 
 private:
-    std::map<std::string, Handler> table_;
+    handler_table_t table_;
 
     template <class HandlerImpl>
     void
     addHandler()
     {
-        assert(table_.find(HandlerImpl::name()) == table_.end());
+        static_assert(HandlerImpl::minApiVer <= HandlerImpl::maxApiVer);
+        static_assert(HandlerImpl::maxApiVer <= RPC::apiMaximumValidVersion);
 
-        Handler h;
-        h.name_ = HandlerImpl::name();
-        h.valueMethod_ = &handle<Json::Value, HandlerImpl>;
-        h.role_ = HandlerImpl::role();
-        h.condition_ = HandlerImpl::condition();
+        assert(!overlappingApiVersion(
```

Comment:
At first blush, this check appears to be redundant. We are performing this check  in the `HandlerTable` constructor, do we need to repeat ourselves?

`addHandler` is not being invoked from any other place in the codebase.

Reply:
I believe we can remove the check in the `HandlerTable` constructor. I verified that the existing unit tests work correctly without this check.

But this is a suggestion, please don't feel its binding to follow it.